editoast/src/views/v2/train_schedule.rs

```diff
@@ -682,41 +683,104 @@ pub async fn project_path(
 #[derive(Debug, Serialize, ToSchema)]
 enum SimulationSummaryResultResponse {
     // Minimal information on a simulation's result
-    #[allow(dead_code)]
     Success {
         // Length of a path
         length: u64,
         // travel time
-        time: f64,
+        time: u64,
         energy_consumption: f64,
     },
-    // Pathfinding not found for a train id
-    #[allow(dead_code)]
-    PathfindingNotFound,
-    // Running time not found or a train id
-    #[allow(dead_code)]
-    RunningTimeNotFound,
+    // Pathfinding fails for a train id
+    PathfindingFailed,
+    // Running time fails or a train id
+    RunningTimeFailed,
 }
 
 /// Retrieve simulation information for a given train list.
 /// Useful for finding out whether pathfinding/simulation was successful.
 #[utoipa::path(
     tag = "train_schedulev2",
+    params(SimulationBatchParams),
     responses(
-        (status = 200, description = "Project Path Output", body = HashMap<Identifier, SimulationSummaryResultResponse>),
+        (status = 200, description = "Project Path Output", body = HashMap<i64, SimulationSummaryResultResponse>),
     ),
 )]
 #[get("/simulation_summary")]
 pub async fn simulations_summary(
-    _db_pool: Data<DbPool>,
-    _redis_client: Data<RedisClient>,
-    _params: QsQuery<ProjectPathParams>,
-    _core_client: Data<CoreClient>,
-    _infra_id: Query<i64>,
-) -> Result<Json<HashMap<Identifier, SimulationSummaryResultResponse>>> {
-    // TO DO
-    // issue:x https://github.com/OpenRailAssociation/osrd/issues/6857
-    Ok(Json(HashMap::new()))
+    db_pool: Data<DbPool>,
+    redis_client: Data<RedisClient>,
+    params: QsQuery<SimulationBatchParams>,
+    core_client: Data<CoreClient>,
+) -> Result<Json<HashMap<i64, SimulationSummaryResultResponse>>> {
+    let query_props = params.into_inner();
+    let infra_id = query_props.infra;
+    let conn = &mut db_pool.clone().get().await?;
+
+    // Retrieve the infra
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || TrainScheduleError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+    let train_ids = query_props.ids;
+    let train_schedule_batch: Vec<TrainSchedule> =
+        TrainSchedule::retrieve_batch_or_fail(conn, train_ids, |missing| {
+            TrainScheduleError::BatchTrainScheduleNotFound {
+                number: missing.len(),
+            }
+        })
+        .await?;
+    // Build a HashMap with train_ids as keys and the corresponding simulation response as values
```

Comment:
I would say you can remove this comment, there already is a comment on the function itself (more probable it's going to be updated if the function evolves, whereas this one might become obsolete more easily).